doc/source/dev/development_workflow.rst

```diff
@@ -200,12 +200,46 @@ sequences. In such cases you may explicitly skip CI by including one of these
 fragments in your commit message:
 
 * ``[skip ci]``: skip all CI
+
+  Only recommended if you are still not ready for the checks to run on your PR
+  (for example, if this is only a draft.)
+
 * ``[skip actions]``: skip GitHub Actions jobs
+
+  `GitHub Actions <https://docs.github.com/actions>`__ is where most of the CI
+  checks are run, including the linter, benchmarking, running basic tests for
+  most architectures and OSs, and several compiler and CPU optimization
+  settings.
+  `See the configuration files for these checks. <https://github.com/numpy/numpy/tree/main/.github/workflows>`__
+
 * ``[skip travis]``: skip TravisCI jobs
+
+  `TravisCI <https://www.travis-ci.com/>`__ will test your changes against
+  Python 3.9 on the PowerPC and s390x architectures.
+  `See the configuration file for these checks. <https://github.com/numpy/numpy/blob/main/.travis.yml>`__
+
 * ``[skip azp]``: skip Azure jobs
+
+  `Azure <https://azure.microsoft.com/en-us/products/devops/pipelines>`__ is
+  where all comprehensive tests are run. This is an expensive run, and one you
+  could typically skip if you do documentation-only changes, for example.
+  `See the main configuration file for these checks. <https://github.com/numpy/numpy/blob/main/azure-pipelines.yml>`__
+
 * ``[skip circle]``: skip CircleCI jobs
+
+  `CircleCI <https://circleci.com/>`__ is where we build the documentation and
+  store the generated artifact for preview in each PR. This check will also run
+  all the docstrings examples and verify their results. If you don't make
```

Comment:
One annoyance is that the refguide checks are actually not run in this step.  Although, it might be nice to try to change that.  @rossbar thoughts?

But... also OK to just put this in for now, the explanations are nice :).

Reply:
Is it not what [this step does](https://github.com/numpy/numpy/blob/46cc9556224ecba0fddd5f78be74289a5ffd64b4/.circleci/config.yml#L77)?

Reply:
(btw this was not the case previously and I was bit by that before as well)

Reply:
Sorry, maybe clearer to say doctests.  The doctests are currently run as part of the azure macos tests.

Reply:
Ahhh you are right? That is even more confusing than I thought... Do you know why this is so? I can do a follow up with that correction.

Reply:
Not sure if there is a particular reason, and if there is it is probably years old.  I guess they need to run on 64bit linux/mac, but beyond that it probably doesn't matter.  It may also be nicer to integrate them into CircleCI (only problem would be if circleCI gets slow, although I suspect you can work-around that with parallel runs)